engine/src/main/java/org/hibernate/search/query/engine/impl/SortConfigurations.java

```diff
@@ -67,7 +67,7 @@ public SortConfiguration(String indexName, Map<Class<?>, List<SortableFieldMetad
 
 			for ( SortField sortField : sort.getSort() ) {
 				// no doc value field needed for these
-				if ( sortField.getType() == SortField.Type.DOC && sortField.getType() == SortField.Type.SCORE ) {
+				if ( sortField.getType() == SortField.Type.DOC || sortField.getType() == SortField.Type.SCORE ) {
```

Comment:
Yes, there is `org.hibernate.search.test.query.sorting.SortTest#testResultOrderedByDocId()`.